src/spatialdata_plot/pl/render.py

```diff
@@ -291,19 +398,109 @@ def _render_points(
+    if method is None:
+        method = "datashader" if len(points) > 10000 else "matplotlib"
+    elif method not in ["matplotlib", "datashader"]:
+        raise ValueError("Method must be either 'matplotlib' or 'datashader'.")
+
+    if method == "datashader":
+        # NOTE: s in matplotlib is in units of points**2
+        px = int(np.round(np.sqrt(render_params.size)))
+
+        extent = get_extent(sdata_filt.points[element], coordinate_system=coordinate_system)
+        x_ext = [min(0, extent["x"][0]), extent["x"][1]]
+        y_ext = [min(0, extent["y"][0]), extent["y"][1]]
+        previous_xlim = ax.get_xlim()
+        previous_ylim = ax.get_ylim()
+        # increase range if sth larger was rendered before
+        if _mpl_ax_contains_elements(ax):
+            x_ext = [min(x_ext[0], previous_xlim[0]), max(x_ext[1], previous_xlim[1])]
+            if ax.yaxis_inverted():  # case for e.g. images
+                y_ext = [min(y_ext[0], previous_ylim[1]), max(y_ext[1], previous_ylim[0])]
+            else:  # case for e.g. labels
+                y_ext = [min(y_ext[0], previous_ylim[0]), max(y_ext[1], previous_ylim[1])]
+        # round because we need integers
+        plot_width = int(np.round(x_ext[1] - x_ext[0]))
+        plot_height = int(np.round(y_ext[1] - y_ext[0]))
+
+        # use datashader for the visualization of points
+        # TODO: what about trans/norm at this point?
+        cvs = ds.Canvas(plot_width=plot_width, plot_height=plot_height, x_range=x_ext, y_range=y_ext)
+
+        color_by_categorical = col_for_color is not None and points[col_for_color].values.dtype == object
+        aggregate_with_sum = None
+        if col_for_color is not None and (render_params.groups is None or len(render_params.groups) > 1):
+            if color_by_categorical:
+                agg = cvs.points(sdata_filt.points[element], "x", "y", agg=ds.by(col_for_color, ds.count()))
+            else:
+                # numerical
+                agg = cvs.points(sdata_filt.points[element], "x", "y", agg=ds.sum(column=col_for_color))
+                # save min and max values for drawing the colorbar
+                aggregate_with_sum = (agg.min(), agg.max())
+        else:
+            agg = cvs.points(sdata_filt.points[element], "x", "y", agg=ds.count())
+
+        color_key = (
+            [x[:-2] for x in color_vector.categories.values]
+            if (type(color_vector) is pd.core.arrays.categorical.Categorical)
+            and (len(color_vector.categories.values) > 1)
+            else None
+        )
+        if color_by_categorical or col_for_color is None:
+            ds_result = ds.tf.shade(
+                ds.tf.spread(agg, px=px),
+                rescale_discrete_levels=True,
+                cmap=color_vector[0][:-2],
+                color_key=color_key,
+                min_alpha=np.min([150, render_params.alpha * 255]),
+            )  # TODO: choose other value than 150 for min_alpha (here and below)?
```

Comment:
Why a different one? Is 150 bad?

Reply:
Not necessarily, but it's an arbitrary value that I chose to make everything better visible...